### DISCUSSION TITLE
Spelling change of Behaviour in directives.ts

### DISCUSSION THREAD

**Discussion by Friday32** [graphql/graphql-js#3526] [Q&A]
What prompted the spelling change from "behaviour" to "behavior" in src/type/directives.ts? This change is causing problems for our program as we use graphql-js in one project and and graphql-java in another.  

Before submitting an issue, I'd like to verify which project should be updated. Keep in mind these repos are used in many countries. It might be best to revert this project to the convention already set.

Change
https://github.com/graphql/graphql-js/commit/82ff6539a5b961b00367ed7d6ac57a7297af2a9a

Current implementation in graphql-java:
https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/Directives.java#L116

Thanks

**Comment by yaacovCR** [comment]
request for more context -- what problem exactly is the alternate spelling in the description causing?

Also, does the spec mandate that description text be identical? Or even provided at all?

**Reply by Friday32** [reply]
It is clear the text came from a common source.  I'd think you'd want keep consistency. 

We have a tool that merges the output of both  generators and there is a conflict because of the difference.  I suppose the tool could  be modified. But, it seems like a bit of work just because someone decided to use an alternate spelling of a word months after using the other.

**Reply by yaacovCR** [reply]
Thanks for providing more context!

Perhaps members of the tsc could tell you which option is "right," but I would still suggest, however, that you consider modifying your tool to handle this situation.

I skimmed the spec and I can't find it mandatory that the built in directives have the same description. I could be missing it! But if this is the case, you could easily encounter this situation again, with differences that are larger and that the various parties involved could be even less likely to eventually harmonize.

Just a few cents. Wishing you luck from more knowledgeable people on the issue. If you don't get a response soon enough, even though it's kind of a narrow issue, you could add it to the working group agenda for this week meeting and then you would be kind of guaranteed some official feedback.

**Comment by Friday32** [comment]
You're right. The best way is updating the merge tool.  What complicates things is our use of a thirdparty tool graphql-codegen that uses graphql-tools/merge.  This repo has a known issue merging descriptions.

What drove us to update versions is a security tool reported security vulnerabilities in an unrelated dependencies of graphql-codegen.

**Comment by saihaj** [comment]
@Friday32 That change was made during our migration to TS I think we made cspell config to enforce everything to US English. It is like `color` in CSS we are just trying to standardize words to US English spellings.

@graphql/tsc thoughts? Should we enforce US English spellings part of reference implementation?

**Reply by benjie** [reply]
Not speaking on behalf of the TSC, only myself, but yes I think consistent spelling using US English is what we should aim for, and changes in descriptions, comments and error messages are suitable for conformance edits (and general spelling/grammar edits) unless explicitly specified in the spec.
